components/vfs/include/components/vfs/filesystem.hpp

```diff
@@ -138,35 +68,31 @@ class RootFileSystem : public FileSystem
 	RootFileSystem(const std::string &base_path = "");
 	virtual ~RootFileSystem() = default;
 
-	virtual bool          folder_exists(const std::string &file_path) override;
-	virtual bool          file_exists(const std::string &file_path) override;
-	virtual StackErrorPtr read_file(const std::string &file_path, std::shared_ptr<Blob> *blob) override;
-	virtual StackErrorPtr read_chunk(const std::string &file_path, const size_t offset, const size_t count, std::shared_ptr<Blob> *blob) override;
-	virtual size_t        file_size(const std::string &file_path) override;
-	virtual StackErrorPtr write_file(const std::string &file_path, const void *data, size_t size) override;
-	virtual StackErrorPtr enumerate_files(const std::string &file_path, std::vector<std::string> *files) override;
-	virtual StackErrorPtr enumerate_folders(const std::string &file_path, std::vector<std::string> *folders) override;
-	virtual void          make_directory(const std::string &path) override;
-
-	StackErrorPtr enumerate_files(const std::string &file_path, const std::string &extension, std::vector<std::string> *files);
-	StackErrorPtr enumerate_files_recursive(const std::string &file_path, const std::string &extension, std::vector<std::string> *files);
-	StackErrorPtr enumerate_folders_recursive(const std::string &file_path, std::vector<std::string> *folders);
+	virtual bool                     folder_exists(const std::string &folder_path) const override;
```

Comment:
It's a small thing.  But worth pointing out should keep override and remove virtual.

Reply:
This is mainly my fault. Although its not needed, i've gotten into the habit of adding virtual to all virtual functions. We should address this in a new PR for the entire branch.